include/revng/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,1083 @@
+    model::Register::rdi_x86_64,  model::Register::rsi_x86_64,
+    model::Register::rbx_x86_64,  model::Register::rbp_x86_64,
+    model::Register::xmm6_x86_64, model::Register::xmm7_x86_64
+  };
+
+  static constexpr auto
+    ReturnValueLocationRegister = model::Register::rax_x86_64;
+};
+
+/// The `__clrcall` Microsoft ABI (can be used for executables of either x86 or
+/// x64 architecture). The parameters are passed using CLR expression stack.
+/// No registers are affected on call.
+///
+/// \note: it's probably a good idea to double-check `CalleeSavedRegisters`,
+/// (the array just lists the defaults), but I'm not sure where I could find
+/// that information.
+template<>
+struct CallingConventionTrait<model::abi::Microsoft_x86_64_clrcall> {
+  static constexpr bool ArgumentsArePositionBased = false;
+  static constexpr bool OnlyStartDoubleArgumentsFromAnEvenRegister = false;
+  static constexpr bool ArgumentsCanBeSplitBetweenRegistersAndStack = false;
+  static constexpr bool UsePointerToCopyForStackArguments = false;
+
+  static constexpr size_t MaximumGPRsPerAggregateArgument = 0;
+  static constexpr size_t MaximumGPRsPerAggregateReturnValue = 0;
+  static constexpr size_t MaximumGPRsPerScalarArgument = 0;
+  static constexpr size_t MaximumGPRsPerScalarReturnValue = 0;
+
+  static constexpr size_t MinimumStackArgumentSize = 8;
+
+  static constexpr std::array<model::Register::Values, 0>
+    GeneralPurposeArgumentRegisters = {};
+  static constexpr std::array<model::Register::Values, 0>
+    GeneralPurposeReturnValueRegisters = {};
+
+  static constexpr std::array<model::Register::Values, 0>
+    VectorArgumentRegisters = {};
+  static constexpr std::array<model::Register::Values, 0>
+    VectorReturnValueRegisters = {};
+
+  static constexpr std::array CalleeSavedRegisters = {
+    model::Register::r12_x86_64,  model::Register::r13_x86_64,
+    model::Register::r14_x86_64,  model::Register::r15_x86_64,
+    model::Register::rdi_x86_64,  model::Register::rsi_x86_64,
+    model::Register::rbx_x86_64,  model::Register::rbp_x86_64,
+    model::Register::xmm6_x86_64, model::Register::xmm7_x86_64
+  };
+
+  static constexpr auto ReturnValueLocationRegister = model::Register::Invalid;
+};
+
+/// The default SystemV x86 ABI
+/// The parameters are only passed using the stack.
+/// \todo: look into the st-return values.
+template<>
+struct CallingConventionTrait<model::abi::SystemV_x86> {
+  static constexpr bool ArgumentsArePositionBased = false;
+  static constexpr bool OnlyStartDoubleArgumentsFromAnEvenRegister = false;
+  static constexpr bool ArgumentsCanBeSplitBetweenRegistersAndStack = false;
+  static constexpr bool UsePointerToCopyForStackArguments = false;
+
+  static constexpr size_t MaximumGPRsPerAggregateArgument = 0;
+  static constexpr size_t MaximumGPRsPerAggregateReturnValue = 0;
+  static constexpr size_t MaximumGPRsPerScalarArgument = 0;
+  static constexpr size_t MaximumGPRsPerScalarReturnValue = 2;
```

Comment:
This means that you can only return up to a single 64-bit scalar, is that right?